### PR TITLE
fix 404 responses

### PR DESCRIPTION
**Pull request by mcfiredrill**
and get !datafruiter working

edited some reply messages

### CODE REVIEW

File: lib/fruitbot/streampusher_api.ex [hunk 1]
```diff
@@ -1,35 +1,68 @@
...
+  def tag_search(query) do
+    case HTTPoison.get!("https://datafruits.streampusher.com/api/archives.json?tags=#{query}") do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        response = Jason.decode!(body)
+        data = response["data"]
+        first_result = Enum.at(data, 0)
 
-        # not sure it seems the API doesn't return a 404 when search result not found, just an empty list
-        %HTTPoison.Response{status_code: 404} ->
+          # uhhh there aren't single pages for achives yet so I'm not sure what link to return...
```
Comment (einariii) [inline lib/fruitbot/streampusher_api.ex:29]: yes i'm seeing that. we can grab a random entry and return a streaming link? or return the artwork thumbnail in the chat? all tags associated with shows, yes?

File: lib/fruitbot/streampusher_api.ex [hunk 2]
```diff
@@ -1,35 +1,68 @@
...
   def user_search(query) do
-  end
+    case HTTPoison.get!("https://datafruits.streampusher.com/api/djs/#{query}.json?name=#{query}") do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        response = Jason.decode!(body)
+        data = response["data"]
+        username = data["attributes"]["username"]
+        url = "https://datafruits.fm/djs/#{username}"
+        "Datafruit found: #{url}"
+      %HTTPoison.Response{status_code: 404} ->
```
Comment (einariii) [inline lib/fruitbot/streampusher_api.ex:10]: the API is still not responding with 404.  i vote to remove this and just have the catch all below `_ -> `

File: lib/fruitbot/streampusher_api.ex [hunk 3]
```diff
@@ -1,35 +1,68 @@
...
+    case HTTPoison.get!("https://datafruits.streampusher.com/api/archives.json?tags=#{query}") do
+      %HTTPoison.Response{status_code: 200, body: body} ->
+        response = Jason.decode!(body)
+        data = response["data"]
+        first_result = Enum.at(data, 0)
 
-        # not sure it seems the API doesn't return a 404 when search result not found, just an empty list
-        %HTTPoison.Response{status_code: 404} ->
+          # uhhh there aren't single pages for achives yet so I'm not sure what link to return...
+      %HTTPoison.Response{status_code: 404} ->
```
Comment (einariii) [inline lib/fruitbot/streampusher_api.ex:30]: see above comment

File: lib/fruitbot/streampusher_api.ex (hunk 1 above)
Comment (mcfiredrill) [reply lib/fruitbot/streampusher_api.ex:29]: Think we can return the direct mp3 link for now

Comment (einariii) [reply lib/fruitbot/streampusher_api.ex:29]: ok leave it to me jefe

File: lib/fruitbot/streampusher_api.ex (hunk 2 above)
Comment (mcfiredrill) [reply lib/fruitbot/streampusher_api.ex:10]: Hmmm seems to be working for for the /djs endpoint
line_20230406_002452

File: lib/fruitbot/streampusher_api.ex (hunk 1 above)
Comment (mcfiredrill) [reply lib/fruitbot/streampusher_api.ex:29]: Cool I'll merge this PR , then you can make another PR to add that if you'd like

Comment (einariii) [reply lib/fruitbot/streampusher_api.ex:29]: excellent!

**Pull request merged**